src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -335,7 +330,7 @@ export function QueryPageContainerComponent(): JSX.Element {
     return (
         <LoadingOrErrorComponent
             status={{ isolateStatus, isolateCountStatus, filterStatus }}
-            dataIsSet={!_.isEmpty(data.ZNData)}
+            dataIsSet={!_.isEmpty(data.uniqueValues)}

```

Comment:
I think it would be better to derive the dataIsSet value from the different status states, so the loading state would not rely on one piece of data.